### PR TITLE
Beware of import qual without symbol

### PR DESCRIPTION
**Pull request by som-snytt**
Test will positify after https://github.com/scala/scala/pull/8617

### CODE REVIEW

File: test/files/neg/t8855.check [hunk 1]
```diff
@@ -0,0 +1,8 @@
+t8855.scala:5: error: stable identifier required, but 42 found.
+    import x._
+           ^
+t8855.scala:6: warning: _root_ in root position of qualifier refers to the root package, not value _root_, which is in scope
+    val g: Int => Int = +
+               ^
```
Comment (hrhino) [inline test/files/neg/t8855.check]: Wait, what is _this_?

**Review by hrhino — APPROVED** [review]
Thanks for cleaning up my mess. I had originally limited the change to `Symbol`/`String`/`Class` literals, but then I decided, hey, maybe someone will want to do exactly that, and didn't bother testing. Shame on me.

File: test/files/neg/t8855.check (hunk 1 above)
Comment (som-snytt) [reply test/files/neg/t8855.check]: It sounds like a Groucho Marx line. I'll take a look, since I obviously didn't look before.

Comment (som-snytt) [reply test/files/neg/t8855.check]: I may add `--no-look` as an alias for `--update-check`.

Comment (som-snytt) [reply test/files/neg/t8855.check]: Apparently the import does import `_root_`, and parser makes the function type `_root_.scala.Function1`. Why does the import do that?

Comment (hrhino) [reply test/files/neg/t8855.check]: Or `--fingers-crossed`. That should also commit, push, and request paulp's review.

**Comment by hrhino** [comment]
I see your related dotty complaint got marked as low priority, what gives? But Martin himself has volunteered to fix it; does that even things out? Personally I'm honored just to see one of my bugs get fixed, by anyone.

File: test/files/neg/t8855.check (hunk 1 above)
Comment (som-snytt) [reply test/files/neg/t8855.check]: It's due to importing from error type, I guess, or maybe no type. BTW, shell history says I didn't actually see the warning; I'd compiled with doti, then changed `Function1` to fat arrow and used :crossed_fingers: option.

Comment (som-snytt) [reply test/files/neg/t8855.check]: "Groucho Marx line": "How a root package got in my pajamas, I'll never know."

**Comment by som-snytt** [comment]
@hrhino I'm also personally honored to get one of your bugs fixed. Thanks for noticing the warning. For erroneous qual `x`, it said `import x._` imports `_root_`, and the warning was that `_root_.scala.Function` was not the root you're rooting for, but the imported one. It was just a coincidence that the NPE had to do with root imports.

**Comment by som-snytt** [comment]
The previous force-push forced too much.

**Pull request merged**